### DISCUSSION TITLE
Formatter Question (custom indentation question)

### DISCUSSION THREAD

**Discussion by goto40** [eclipse-langium/langium#1542] [Q&A] [answered]
We are implementing a formatter are quite happy to be able to do the following (":" are all aligned at the same level)

```
A {
    element name1         : info ( a, b ,c )
    another element name2 : info ( a, b )
}
```

**We are not sure if we can achieve the following (is this possible somehow?)**
```
A {
    element name1         : info
                          (
                              a,
                              b,
                              c
                          )
    another element name2 : info
                          (
                              a,
                              b
                          )
}
```

We already can do the following ("(" in a new line, but not aligned with ":")
```
A {
    element name1         : info
    (
	    a,
	    b,
	    c
    )
    another element name2 : info
    (
	    a,
	    b
    )
}
```
)

**Comment by msujew** [comment, marked as answer]
Hey @goto40,

You probably won't get this behavior out of the public formatter API. You would need to override how the edits are generated in the abstract class.

I'll take note of that requirement for my upcoming refactoring of the formatter :)

**Reply by goto40** [reply]
It would be nice if we could define "named tabulator positions", which should ne aligned in consecutive lines.
(`append(oneSpace())`, ..., `append(tabulator("tabulator1")`)
```
A {
  a <tabulator1>: info123 <tabulator2> (a,b)
  another_a <tabulator1>: info1 <tabulator2> (a,b)
  a2 <tabulator1>: info23 <tabulator2> (a,b)
}
```
to yield
```
A {
  a         : info123 (a,b)
  another_a : info1   (a,b)
  a2        : info23  (a,b)
}
```

**Reply by goto40** [reply]
What do you think (concerning your requirements ;-) )?

**Reply by msujew** [reply]
Sounds good. I'll see how to fit that into a easy-to-use API :)

**Reply by goto40** [reply]
Any news concerning this API question?
